examples/component/http-keyvalue-bus/main.go

```diff
@@ -0,0 +1,86 @@
+//go:generate go run go.bytecodealliance.org/cmd/wit-bindgen-go generate --world hello --out gen ./wit
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	// A lightweight, high performance HTTP request router.
+	"github.com/julienschmidt/httprouter"
+
+	// For the keyvalue capability, we're using bindings for the wasi:keyvalue/store and /atomics interfaces.
+	atomics "http-keyvalue-bus/gen/wasi/keyvalue/atomics"
+	store "http-keyvalue-bus/gen/wasi/keyvalue/store"
+
+	// The wasmcloud:bus/lattice interface enables us to set the operative link definition for an interface.
+	lattice "http-keyvalue-bus/gen/wasmcloud/bus/lattice"
+
+	// The cm module provides types and functions for interacting with the WebAssembly Component Model.
+	cm "go.bytecodealliance.org/cm"
+
+	// These wasmCloud modules enable us to write more idiomatic Go when using wasi:http and wasi:logging.
+	"go.wasmcloud.dev/component/log/wasilog"
+	"go.wasmcloud.dev/component/net/wasihttp"
+)
+
+func init() {
+	// Establishes the routes for the application.
+	router := httprouter.New()
+	router.GET("/", indexHandler)
+	router.GET("/:kv/:name", kvHandler)
+	wasihttp.Handle(router)
+}
+
+func indexHandler(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	fmt.Fprintf(w, "Make a request to /redis/<name> or /nats/<name>\n")
+}
+
+func kvHandler(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	logger := wasilog.ContextLogger("kvHandler")
+
+	// Return if the kv parameter doesn't specify one of our KV providers.
```

Comment:
IMO: This is not wrong/bad but think it was based on our Rust implementation and would be simpler to provide 2 wadm yamls instead of runtime linking. one for redis and one for nats, both going thru `default` link.

Runtime linking is nuanced and we don't use it very often.  [codesearch](https://github.com/search?q=repo%3AwasmCloud%2FwasmCloud+set_link_name+path%3A%2F%5Eexamples%5C%2F%2F&type=code)
With TinyGo .33 and large wasm-collider https://github.com/lxfontes/lwc/blob/main/probe/main.go#L35 we saw situations where repointing the link locked up the component. slightly different situation due to long tail-calls but still strange.

It's also one of those 🤔 features that might change/go-away as wasip3 comes along.